### DISCUSSION TITLE
404 routing fails

### DISCUSSION THREAD

**Discussion by eliasu** [studio1902/statamic-peak#388] [Q&A] [answered]
Hi, 

I'm using Peak and I'm wondering why my 404 Page does not work. 

I obvioulsy use the Globals/Configuration Settings to setup the 404:
![Bildschirmfoto 2024-03-25 um 13 02 42](https://github.com/studio1902/statamic-peak/assets/11287110/8a8fe8ec-5514-4b44-ad98-59696ebac9ee)

But if a page is not found, it will return an **500 Intern Error**
![Bildschirmfoto 2024-03-25 um 13 03 51](https://github.com/studio1902/statamic-peak/assets/11287110/777acd84-69a7-4efc-a719-dbe438a2ae27)

I think I messed up my routing or something else, because usually this works flawlessly.
My question is, if you could guide me, how to fix this?  Where could I start to look for errors? How to use the Debugbar or any other Debug Tool for this?

Any help is appreciated! 
Thanks.

**Comment by robdekort** [comment]
Do you have the tools addon installed still?

**Reply by eliasu** [reply]
I guess so? Do I miss something?

![Bildschirmfoto 2024-03-25 um 14 36 02](https://github.com/studio1902/statamic-peak/assets/11287110/045be592-49b9-4925-b01f-97e34b5a7e54)

**Comment by robdekort** [comment]
Anything in the log?

**Reply by eliasu** [reply]
Which log do you mean? Console? "Admin log"? Both of those, say nothing …
Maybe Debugbar helps, but I don't know where to look really … ?

**Reply by robdekort** [reply]
If there's a 500 error there should be something in your `laravel.log`.

**Comment by eliasu** [comment]
Ok it gets weirder down the rabbit hole :)

So, I found out, that the problem was a "faulty" tag which i wrote. It works everywhere, except for the 404 page. 

This code inside my tag:
```php
$page = $this->context->get('page');
debugbar()->info($page);
```

returns something like this on every page that I visit. 
![Bildschirmfoto 2024-03-25 um 16 34 53](https://github.com/studio1902/statamic-peak/assets/11287110/fb1a4df6-7e5f-4c5a-8a77-c8a05ad2d73a)

But it will return `null` on the 404 page. That causes the problem. I cant write a fix for that, but do you have a clue why this is happening?

**Reply by robdekort** [reply, marked as answer]
I think cuz a page like a 404 isn't actually a page but a route. So you're requesting stuff that doesn't exist. You could perhaps wrap it in `{{ unless response_code === '404' }}`.

**Reply by eliasu** [reply]
Leets goo! 

Fixed it with ` if (!$page)` and return a standard fallback value.
Well thanks for putting me in the right direction. 🙏

Bye!
